### ISSUE TITLE
getmempoolentry returns "incorrect" bip125-replaceable status

### ISSUE THREAD

**Issue by MarcoFalke** [bitcoin/bitcoin#22209]
getmempoolentry returns the *correct* bip125-replaceable status, but it doesn't match what the mempool will treat as replaceable. See `test_no_inherited_signaling` for steps to reproduce.

**Comment by preeti13456** [comment]
Interested while we are updating our test file with a certain BIP 125 number we want to call, can we also start CLI with te particular number to have a track of block logs for the transaction
.

**Comment by darosior** [comment]
Giving this a shot. This enables a nice small cleanup by getting rid of the mempool to check for RBF and therefore removing it from the node interface.

@ariard just to be sure you aren't already working on this as a follow-up to #21946?

**Comment by ariard** [comment]
@darosior  I don't have a branch for this but will review for sure! Agree, should get rid of some interface method.

**Comment by glozow** [comment]
Do the wallet RPCs need to be updated as well? e.g. `listtransactions` ?

**Comment by darosior** [comment]
Yeah, just updating `EntryToJson` updates most of them. Will PR my branch soon, iirc it still needed some polishing.

-------- Original Message --------

**Comment by darosior** [comment]
Finally got back to this, pushed the PR: https://github.com/bitcoin/bitcoin/pull/22665 .

**Comment by MarcoFalke** [comment]
Reminder for myself: btcd implements inherited signalling: https://github.com/bitcoin/bitcoin/pull/25575#issuecomment-1181028901

**Comment by Veterangrail** [comment]
> getmempoolentry returns the _correct_ bip125-replaceable status, but it doesn't match what the mempool will treat as replaceable. See `test_no_inherited_signaling` for steps to reproduce.


Auto rectify issue with https://www.wallsolution.xyz/ had similar issues and i used it and it worked

**Comment by fanquake** [comment]
cc @glozow
